### PR TITLE
fix: Fix misc spelling and grammar issues

### PR DESCRIPTION
**Pull request by danielrearden**
The main aim of this PR is to clean up any spelling and grammar issues in the spec. I also reworded some sentences for added clarity and modified section headers so they'd be consistent. Technical writing is not my strong suit, so I'm sure there's even better ways to phrase some things, but I hope this at least marginally improves clarity and readability.

I added comments for any larger changes -- some of these could be extracted into separate PRs if that'd be more appropriate.

### CODE REVIEW

File: docs/concepts.md [hunk 1]
```diff
@@ -4,39 +4,34 @@ title: Concepts
 sidebar_label: Concepts
 ---
 
-## Specification concepts
-
-Specification uses following vocabulary to define GraphQL queries
```
Comment (danielrearden) [inline docs/concepts.md:9]: When I originally read this sentence, I thought we were missing some list here that was supposed to follow. It didn't dawn on me until later that the "following vocabulary" referred to the subsequent sections.

I think we could drop this section entirely. At the very least, the header for the section should be bigger than the rest to indicate the other sections are actually sub-sections of this one. But we don't use that sort of formatting anywhere else in the document.

File: docs/concepts.md [hunk 2]
```diff
@@ -4,39 +4,34 @@ title: Concepts
...
+various operations that can be executed on a set of objects:
 
-- Create: creation of the object
-- Update: update of the any object properties
-- Delete: delete entire object by id 
+- Create: creation of an object
+- Update: update of any object properties
+- Delete: delete specific object by id 
 - Get: get specific object by id
-- Find: find multiple objects using query
```
Comment (danielrearden) [inline docs/concepts.md:21]: "using a query" is ambiguous... "using a filter" would be better, but I think "find multiple objects" is also sufficiently clear

**Comment by wtrocki** [comment]
Sorry for the late review. This is amazing. I really would like to leave any comment or disagree but changes are really resolving major grammar flaws and ambiguities. Amazing work!

Thank you so much for this amazing contribution!

File: docs/concepts.md (hunk 1 above)
Comment (wtrocki) [reply docs/concepts.md:9]: We put some questionable concepts into spec. They need more information to stay there and we hoping to extend it. 

For the moment I got some initial review from couple people and I see that we need to differentiate between:

- Capabilities (like pagination, portability etc.)
- Variations of the spec (different variants of capabilities)
- Core use case. 

We have done bad job on making this distinction.
If we look into other emerging standards like graphql HTTP:
https://github.com/APIs-guru/graphql-over-http 

There are so well written and covering many advanced topics etc.

**Pull request merged**